keps/sig-cloud-provider/0002-cloud-controller-manager.md

```diff
@@ -434,9 +774,58 @@ Community norms.
 
 ## Alternatives
 
+### Dedicated Cloud Provider Executable Alternatives 
 One alternate to consider is the use of a side-car. The cloud-interface in tree could then be a [GRPC](https://github.com/grpc/grpc-go)
 call out to that side-car. We could then leave the Kube API Server, Kube Controller Manager and Kubelet pretty much as is.
 We would still need separate repos to hold the code for the side care and to handle cluster setup for the cloud provider.
 However we believe that different cloud providers will (already) want different control loops. As such we are likely to need
 something like the cloud controller manager anyway. From the perspective it seems easier to centralize the effort in that
 direction. In addition it should limit the proliferation of new processes across the entire cluster.
+
+### Staging Alternatives
+
+#### Git Filter Branch
+
+One possible alternative is to make use of a Git Filter Branch. This would be an alternative to the Staging mechanism. 
+This has issues such as not giving K8s/K8s developers any indications of what the dependencies various 
+K8s/Cloud-providers have. Without that information it becomes very easy to accidentally break various cloud providers 
+and time you change dependencies in the K8s/K8s repo. With staging the dependency line is simple and automatically 
+enforced. Things in Staging are not allowed to depend on things outside of Staging. If you want to add such a 
```

Comment:
Two things that in my opinion could be slightly clearer: 1. Is the git filter branch proposal to just move the code straight to the cloud provider repository?  2. The mechanism which enforces "Things in Staging are not allowed to depend on things outside of Staging" (maybe a link would suffice)

Reply:
Added